### PR TITLE
Fixed long locus ID issue in GB parser

### PR DESCRIPTION
**Pull request by tcjaffe**
Also added support for missing linear/circular token like what we see in the GenBank sample file. Attempts to address the following issue:
 
 https://github.com/biotite-dev/biotite/issues/486

### CODE REVIEW

File: src/biotite/sequence/io/genbank/metadata.py [hunk 1]
```diff
@@ -64,12 +64,45 @@ def get_locus(gb_file):
...
+
+    # The third field *should* be the molecular type 
+    # but sometimes this is missing
+    # NOTE: remember that fields[2] is the unit for length, 
+    #       eg bp or aa, so we move to fields[3] here.
+    if fields[3] not in ('linear', 'circular'):
+        mol_type = fields[3]
+        next_idx = 4
+    else:
+        mol_type = ''
```
Comment (tcjaffe) [inline src/biotite/sequence/io/genbank/metadata.py:84]: note: i set mol_type to '' here instead of None in order to comply with existing unit tests. i'm happy to switch this to None and adjust existing references if that's better

File: src/biotite/sequence/io/genbank/metadata.py [hunk 2]
```diff
@@ -64,12 +64,45 @@ def get_locus(gb_file):
...
+        next_idx = 3
+
+
+    # The next field should be the token 'linear' or 'circular', 
+    # but sometimes this is missing
```
Comment (tcjaffe) [inline src/biotite/sequence/io/genbank/metadata.py:89]: note: i don't handle the edge case where both mol_type and is_circular are missing because i haven't yet seen that edge case. i can look into to implementing this, though, if it seems worthwhile

File: src/biotite/sequence/io/genbank/metadata.py [hunk 3]
```diff
@@ -64,12 +64,45 @@ def get_locus(gb_file):
...
+    if 'linear' == fields[next_idx]:
+        is_circular = False
+        next_idx += 1
+    elif 'circular' == fields[next_idx]:
+        is_circular = True
+        next_idx += 1
+    else:
+        is_circular = False
+
+    # The next field is the division, an all caps 3 letter token
```
Comment (tcjaffe) [inline src/biotite/sequence/io/genbank/metadata.py:99]: note: i don't actually assert anything against the format here, i just assume it's the division

File: src/biotite/sequence/io/genbank/metadata.py [hunk 4]
```diff
@@ -64,12 +64,45 @@ def get_locus(gb_file):
...
+    division = fields[next_idx]
+    next_idx += 1
+
+    # The last field is a date in the format DD-M-YYYY
+    date = fields[next_idx]
```
Comment (tcjaffe) [inline src/biotite/sequence/io/genbank/metadata.py:104]: same as above re: not asserting against the format

File: tests/sequence/data/missing_linear_vs_circular.gb [hunk 5]
```diff
@@ -0,0 +1,165 @@
+LOCUS       SCU49845     5028 bp    DNA             PLN       21-JUN-1999
```
Comment (tcjaffe) [inline tests/sequence/data/missing_linear_vs_circular.gb:1]: this is actually the sample file supplied by GenBank when describing the gb flat file format. it wasn't working before.

**Review by padix-key — COMMENTED** [review]
Thanks for implementing the fix. I have appended a few minor comments. If you like you, may also add yourself to `CONTRIB.rst`

**Comment by tcjaffe** [comment]
> Thanks for implementing the fix. I have appended a few minor comments. If you like you, may also add yourself to `CONTRIB.rst`

Thank you! I'll apply the feedback and get an update PR up soon.

File: src/biotite/sequence/io/genbank/metadata.py (hunk 2 above)
Comment (tcjaffe) [reply src/biotite/sequence/io/genbank/metadata.py:118]: Added. This was a little tricky for me because I couldn't find a set list of molecular types to check against. I ended up checking against a published list of divisions (see https://www.ncbi.nlm.nih.gov/genbank/samplerecord/#GenBankDivisionA) instead.

Even that didn't quite work, though, because the list they publish in the sample record documentation was actually missing a division, CON, which was fortunately covered in an existing unit test. I checked the full list of divisions currently supported by their search functionality [here](https://www.ncbi.nlm.nih.gov/nuccore/advanced) by selecting Division and clicking "Show index list" to make sure that the list I coded up here is complete.

So long story short, I'm checking against a list of divisions currently supported by GenBank but I'm worried that list may change again in the future making this code somewhat brittle.

File: src/biotite/sequence/io/genbank/metadata.py [hunk 6]
```diff
@@ -64,12 +64,75 @@ def get_locus(gb_file):
...
+    # by units (eg 1224 aa)
+    length = int(fields[1])
+
+    # The third field *should* be the molecular type 
+    # but sometimes this is missing.  This gets tricky
+    # because sometimes the next field, circular/linear,
+    # is missing, too. The field after that, division,
+    # is a 3 letter all caps token. Unfortunately, mol_type
+    # is also often a 3 letter all caps token (eg DNA)! 
+    # Fortunately, GenBank publishes the set list of divisions,
```
Comment (padix-key) [inline src/biotite/sequence/io/genbank/metadata.py:82]: Can you link the source here?

File: src/biotite/sequence/io/genbank/metadata.py [hunk 7]
```diff
@@ -64,12 +64,75 @@ def get_locus(gb_file):
...
+    # so we can check against that set when determining whether
+    # the current token represents the molecular type.
+    # NOTE: remember that fields[2] is the unit for length, 
+    #       eg bp or aa, so we move to fields[3] here.
```
Comment (padix-key) [inline src/biotite/sequence/io/genbank/metadata.py:86]: This comment is probably supposed to go below the list of divisions, right?

File: src/biotite/sequence/io/genbank/metadata.py (hunk 2 above)
Comment (padix-key) [reply src/biotite/sequence/io/genbank/metadata.py:118]: I agree, but this is still our best effort, so in my opinion it is OK for now. In the future it is easy to add further divisions to the list

File: src/biotite/sequence/io/genbank/metadata.py [hunk 8]
```diff
@@ -14,7 +14,7 @@
 
 from ....file import InvalidFileError
 from .file import GenBankFile
-
+import re
```
Comment (padix-key) [inline src/biotite/sequence/io/genbank/metadata.py:17]: You are not using the `re` module anywhere, right?

**Review by padix-key — CHANGES_REQUESTED** [review]
Just a few aesthetic suggestions are remaining from my side, otherwise the PR looks very good! Due to your comprehensive changes in `src/biotite/sequence/io/genbank/metadata.py`, I would propose you add yourself the author field too: `__author__ = "Patrick Kunzmann, Natasha Jaffe"`

**Comment by tcjaffe** [comment]
Thanks again for the review, @padix-key . I think I made all the requested changes. Can you take another look when you have time?

**Review by padix-key — APPROVED** [review]
Thanks! I will merge this PR. The failing test is unrelated.
